apps/web/src/views/Swap/TwapAndLimit.tsx

```diff
@@ -0,0 +1,229 @@
+      const otherCurrencyId = isInput ? outputCurrencyId : inputCurrencyId
+      const newCurrencyId = currencyId(newCurrency)
+      if (newCurrencyId === otherCurrencyId) {
+        replaceBrowserHistory(isInput ? 'outputCurrency' : 'inputCurrency', oldCurrencyId)
+      }
+      replaceBrowserHistory(isInput ? 'inputCurrency' : 'outputCurrency', newCurrencyId)
+    },
+    [onCurrencySelection, warningSwapHandler, inputCurrencyId, outputCurrencyId],
+  )
+
+  const { isChartSupported, isChartDisplayed, setIsChartDisplayed } = useContext(SwapFeaturesContext)
+
+  const toggleChartDisplayed = () => {
+    setIsChartDisplayed?.((currentIsChartDisplayed) => !currentIsChartDisplayed)
+  }
+
+  const onSrcTokenSelected = useCallback(
+    (token: Currency) => {
+      handleCurrencySelect(true, token)
+    },
+    [handleCurrencySelect],
+  )
+
+  const onDstTokenSelected = useCallback(
+    (token: Currency) => {
+      handleCurrencySelect(false, token)
+    },
+    [handleCurrencySelect],
+  )
+
+  return (
+    <>
+      <div>
+        <Swap.CurrencyInputHeader
+          title={
+            <Flex alignItems="center" width="100%" justifyContent="space-between">
+              <Swap.CurrencyInputHeaderTitle>{limit ? 'LIMIT' : 'TWAP'}</Swap.CurrencyInputHeaderTitle>
+              {isChartSupported && (
+                <ColoredIconButton
+                  onClick={() => {
+                    toggleChartDisplayed()
+                  }}
+                  variant="text"
+                  scale="sm"
+                  data-dd-action-name="Price chart button"
+                >
+                  {isChartDisplayed ? (
+                    <ChartDisableIcon color="textSubtle" />
+                  ) : (
+                    <ChartIcon width="24px" color="textSubtle" />
+                  )}
+                </ColoredIconButton>
+              )}
+            </Flex>
+          }
+          subtitle={<></>}
+        />
+      </div>
+
+      <FormContainer>
+        <PancakeTWAP
+          ConnectButton={ConnectWalletButton}
+          connectedChainId={chainId}
+          account={account}
+          limit={limit}
+          usePriceUSD={useUsd}
+          useTrade={useBestTrade}
+          dappTokens={tokens}
+          isDarkTheme={resolvedTheme === 'dark'}
+          srcToken={inputCurrencyId}
+          dstToken={outputCurrencyId}
+          useTokenModal={useTokenModal}
+          onSrcTokenSelected={onSrcTokenSelected}
+          onDstTokenSelected={onDstTokenSelected}
+          ordersId={!isDesktop ? 'twap-order-history-mobile' : undefined}
+          nativeToken={native}
+          connector={connector}
+        />
+      </FormContainer>
+    </>
+  )
+}
+
+export const OrderHistory = () => {
+  const { isDesktop } = useMatchBreakpoints()
+
+  return (
+    <BodyWrapper style={{ maxWidth: 'unset', marginTop: isDesktop ? 0 : 20 }}>
+      <Wrapper id="swap-page" style={{ padding: 0 }}>
+        <Orders />{' '}
```

Comment:
Can we remove this space here?
```suggestion
        <Orders />
```